### PR TITLE
pingid: client is now a subclass of the PluginSDK provided MFAClient …

### PR DESCRIPTION
**Pull request by ddomeb**
…class

Signed-off-by: Dome Darazs <dome.darazs@oneidentity.com>

### CODE REVIEW

File: lib/pingid.py [hunk 1]
```diff
@@ -39,6 +52,90 @@ def __init__(self, url, alias, token, key):
...
+        params.update({"deviceId": device_id} if device_id else {})
+        params.update({"authType": "OTP"} if request_otp else {})
+
+        params.update({"formParameters": {}})
+        params["formParameters"].update({"sp_name": sp_name} if sp_name else {})
+        params["formParameters"].update({"sp_logo": sp_logo} if sp_logo else {})
+
+        return self._call_api("/rest/4/authonline/do", username, params)
+
+    def auth_offline(self, username, otp, session_id):
```
Comment (gorosz) [inline lib/pingid.py]: make this protected

File: lib/pingid.py [hunk 2]
```diff
@@ -39,6 +52,90 @@ def __init__(self, url, alias, token, key):
...
+        response = self.auth_online(mfa_identity, self.device_id,
+                                            sp_name=self.push_display_text, sp_logo=self.logo_url)
+        return self._get_verdict(response)
+
+    def auth_start(self, username, device_id=None):
+        params = {}
+        params.update({"deviceId": device_id} if device_id else {})
+        return self._call_api("/rest/4/startauthentication/do", username, params)
+
+    def auth_online(self, username, device_id=None, request_otp=False, sp_name=None, sp_logo=None):
```
Comment (gorosz) [inline lib/pingid.py]: make this protected

File: lib/pingid.py [hunk 3]
```diff
@@ -39,6 +52,90 @@ def __init__(self, url, alias, token, key):
...
+        response = self.auth_offline(mfa_identity, otp, session_id)
+
+        return self._get_verdict(response)
+
+    def push_authenticate(self, mfa_identity):
+        response = self.auth_online(mfa_identity, self.device_id,
+                                            sp_name=self.push_display_text, sp_logo=self.logo_url)
+        return self._get_verdict(response)
+
+    def auth_start(self, username, device_id=None):
```
Comment (gorosz) [inline lib/pingid.py:71]: make this protected

File: lib/pingid.py [hunk 4]
```diff
@@ -1,35 +1,48 @@
...
 
 
-class Client:
+class Client(MFAClient):
     API_VERSION = 4.9
     JWS_ALG = "HS256"
 
     @classmethod
-    def from_config(cls, configuration):
+    def from_config(cls, configuration, device_id, push_display_text):
```
Comment (gorosz) [inline lib/pingid.py:27]: Can default params use for the new one: device_id=None?

**Review by gorosz — CHANGES_REQUESTED** [review]

**Comment by gasparb16** [comment]
LGTM

File: lib/pingid.py (hunk 3 above)
Comment (ddomeb) [reply lib/pingid.py:71]: this must be called explicitly from plugin.py (line 74) so I'd rather leave it public.

**Review by gorosz — APPROVED** [review]

**Pull request merged**